### DISCUSSION TITLE
List of created `Base`s

### DISCUSSION THREAD

**Discussion by schaumb** [deta/cloud-docs#238] [Help] [answered]
How can I query from [python API](https://docs.deta.sh/docs/base/py_tutorial/) the `deta` currently exists `Base`s?

**Comment by abdelhai** [comment]
a good way to do that is to use `fetch()` on a base and if the response is empty, then base doesn't exists (if a base is empty, then it practically doesn't exist)

**Reply by schaumb** [reply]
With that method, I can check only one base existence, but not list all of `Base`. 

This means currently no practical way to list them from API if I understand correctly, only with http querying `https://web.deta.sh/home/<name>/<project>/bases` Deta UI, or I need to iterate all word (from 'a' to 'Z' then 'aa' to 'ZZ' etc).
Is this correct?

**Comment by jnsougata** [comment, marked as answer]
Currently that's not possible. If you have list of possible base names you have to make query for each base name and check if that exists.
